### DISCUSSION TITLE
Toggle direct permissions?

### DISCUSSION THREAD

**Discussion by PenguinDetective** [spatie/laravel-permission#1860] [Q&A]
How would I go about checking if a permission is assigned to a role or a user directly? I thought about adding a flag, `is_direct` (or any suggestions on a better name) to the permission and only if this boolean value is true, then allow to assign it directly to a user but not a role, and vice versa.

**Comment by erikn69** [comment]
https://github.com/spatie/laravel-permission/blob/2512d5a22aaae0bab8d62a02e502a603776da9da/src/Traits/HasPermissions.php#L275
https://github.com/spatie/laravel-permission/blob/2512d5a22aaae0bab8d62a02e502a603776da9da/src/Traits/HasPermissions.php#L262
You could make your own functions based on those
